SETUP.md

```diff
@@ -0,0 +1,78 @@
+# Contributing
+
+Thank you for considering contributing to the R Markdown C
+
+These instructions describe the steps and system requirements needed to build this book. 
+
+When contribution to the book, it is a good practice to build it locally to ensure that any modifications that you have made worked as expected. However, please do not include the built HTML or PDF in your pull request. It is easier to merge in plain text R, Markdown, and R Markdown files and to only reknit the book on the master branch. 
+
+## System Requirements
+
+### R Packages
+
+The following R packages are required to build to book. These are the same as those which are loaded in the `index.Rmd` file, plus `bookdown` and `tinytex`.
+
+```
+reqd_pkgs <-
+c('remotes', 'webshot', 'DiagrammeR', 'stargazer', 'gt', 'flextable', 'formatR',
+  'equatiomatic', 'gifski', 'magick', 'pdftools', 'tikzDevice', 'xaringan',
+  'kableExtra', 'blogdown', 'jsonlite', 'nomnoml', 'bookdown', 'tinytex')
+```
+
+Additionally, the following packages are referenced in the book. If you do not have them installed, the book will still build, but there references will be incomplete and you will receive numerous messages in the console of the form "pandoc-citeproc: reference R-{package name} not found". 
+
+```
+refd_pkgs <- 
+  c('animation', 'blastula', 'blogdown', 'bookdown','broom', 'Cairo', 'cairoDevice', 'condformat',
+    'dagitty', 'diagram', 'DiagrammeR', 'distill', 'downloadthis', 'DT', 'ezknitr', 'flair',
+    'flexdashboard', 'flextable', 'formatR', 'formattable', 'gganimate', 'ggdag', 'ggplot2', 
+    'gifski', 'googledrive', 'govdown', 'gt', 'gtsummary', 'here', 'huxtable', 'kableExtra', 
+    'knitcitations', 'knitr', 'learnr', 'magick', 'nomnoml', 'officedown', 'officer', 'pagedown',
+    'pander', 'pixiedust', 'pkgdown', 'printr', 'r2d3', 'reactable', 'reticulate', 'revealjs', 'rgl',
+    'rhandsontable', 'rmarkdown', 'rmdformats', 'roxygen2', 'rsconnect', 'rticles', 'sass', 'spelling', 
+    'stargazer', 'styler', 'svglite', 'tables', 'tangram', 'tikzDevice', 'tinytex', 'tufte', 'usethis',
+    'webshot', 'workflowr', 'xaringan', 'xfun', 'xtable', 'yaml', 'ztable')
+```
+
+You may install packages with `install.packages()` or with the **pak** package's `pkg_install()` function, i.e.:
+
+```
+pak::pkg_install(reqd_pkgs)
+```
+
+Note the `pkg_install()` will only install missing packages (and not update current packages) unless you set `upgrade = TRUE`. It is generally a good idea to work with the latest version of packages when preparing a contribution or, at minimum if you receive errors while building, upgrading your packages is a good first step in debugging.
+
+### LaTeX Packages
```

Comment:
The only dependency required to build the HTML book are these one, installed automatically if you don't have them already
https://github.com/yihui/rmarkdown-cookbook/blob/473b21afca0bc063c68747945c2b1aba75edd24f/11-chunk-options.Rmd#L478-L479

You just need Tinytex. 

Should we do like asymptote and skipped if TinyTeX not installed ?

Reply:
I like this approach for Asymptote since we found, as Windows users, that installing it didn't help anyway. However, I wonder if instead for `tinytex` we should automate it like the other installs? 

I notice `tinytex` is not listed in the packages in the `index.Rmd` and in `01-installation.Rmd`, the chunk which discusses `tinytex::install_tinytex()` is set to `eval = FALSE` and not conditional. 

Shall I update those two files to include to install `tinytex` (the R package) and then evaluate the chunk to install TinyTex if not found (e.g. `!nzchar(tinytex::tinytex_root())`)?

Reply:
About the package not being listed, I believe this is because it is a strong dependency of **rmarkdown**, so it will be available for sure in such project. but we could add it if we want to have an exhaustive list of packages called in the book. 

For TinyTeX, the distribution, it is not a requirement per-se. if you have TeX Live for example, this should work. 
`tinytex::_tinytex_root()` will only check for TinyTeX not TeX Live I think. Should we consider a TeX distribution a prerequisite of the book to install manually (and TinyTeX is a good option) instead of forcing installation of TinyTeX ? Like asymptote on linux. Only the required TeX packages could be installed automatically. 

But as authors of the book, I am ok with using TinyTeX. So I don't have strong opinion on that, Do as you think it is easier. At the end, this is mainly for us so that it is easy to build the book from a fresh environment.